app/controllers/api/professor/rooms_controller.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+module Api
+  module Professor
+    class RoomsController < ProfessorController
+      before_action :find_exam_and_course
+      before_action :require_prof_reg
+
+      def index
+        room_regs = @exam.registrations.group_by(&:room)
+        render json: {
+          unassigned: @exam.unassigned_students.map do |s|
+            serialize_student s
+          end,
+          rooms: @exam.rooms.map do |room|
+            regs = room_regs[room] || []
+            serialize_room_regs room, regs
+          end,
+          sections: @exam.course.sections.map do |section|
+            {
+              id: section.id,
+              title: section.title,
+              students: section.students.map do |student|
+                serialize_student student
+              end
+            }
+          end
+        }
+      end
+
+      def update_all
+        body = params.permit(unassigned: [], rooms: {})
+        body[:unassigned].each do |id|
+          # TODO: remove registration
```

Comment:
@blerner should we allow the professor to leave students unassigned? If so, should we delete those registrations? or maybe mark them inactive?